docs/src/components/templates.tsx

```diff
@@ -126,15 +122,25 @@ function useLastTruthyValue<T>(val: T): T {
 }
 
 export default function Templates() {
-  const router = useRouter();
-  const { template } = router.query;
-  const templateMatch = templates.find(t => t.id === template);
+  const searchParams = useSearchParams();
+  const urlTemplateId = searchParams?.get('template') || '';
 
+  const [currentTemplateId, setCurrentTemplateId] = useState(urlTemplateId);
+
+  const templateMatch = templates.find(t => t.id === currentTemplateId);
   const lastTemplate = useLastTruthyValue(templateMatch);
 
   const onClose = () => {
-    const { template: _template, ...restQuery } = router.query;
-    router.push({ query: restQuery }, undefined, { scroll: false });
+    // Remove the template search param
+    const queryParams = new URLSearchParams(searchParams || '');
+    queryParams.delete('template');
+    history.pushState(
+      {},
+      '',
+      Array.from(queryParams.entries()).length ? `?${queryParams}` : '/'
```

Comment:
Could do `queryParams.size ? ...` but since we're throwing this away at some point, probably not worth it?

Reply:
Yessss so much better. `size` didn't come up in intellisense. That's what I get being too lazy to open the docs 🫣

Reply:
Ah, that's why it doesn't come up in intellisense:
<img width="798" alt="image" src="https://github.com/Thinkmill/keystatic/assets/5933897/7f2647d3-91d4-4f72-ab4e-fdba1bd0e48a">

https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/size#browser_compatibility

Thanks to our TS CI step for alerting me to that 😅

Reply:
Ahh interesting - TIL. iOS Safari yikes. I've defs used this places I shouldn't 😅

Reply:
So do you know if there's a VS extension/config I can apply the same CI rule?